streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -272,15 +302,19 @@ private void maybeCompleteRestoration(final StreamTask task,
             if (restoredChangelogs.containsAll(taskChangelogPartitions)) {
                 task.completeRestoration(offsetResetter);
                 log.debug("Stateful active task " + task.id() + " completed restoration");
-                addTaskToRestoredTasks(task);
+                addToRestoredTasks(task);
                 updatingTasks.remove(task.id());
-                if (onlyStandbyTasksLeft()) {
-                    changelogReader.transitToUpdateStandby();
-                }
+                transitToUpdateStandbysIfOnlyStandbysLeft();
             }
         }
 
-        private void addTaskToRestoredTasks(final StreamTask task) {
+        private void transitToUpdateStandbysIfOnlyStandbysLeft() {
+            if (onlyStandbyTasksLeft()) {
+                changelogReader.transitToUpdateStandby();
+            }
+        }
```

Comment:
We missed to switch to updating standby tasks when all active tasks are removed or they failed.